src/main/java/org/extism/sdk/chicory/Manifest.java

```diff
@@ -65,6 +67,11 @@ public Options withWasi(WasiOptions wasiOptions) {
             return this;
         }
 
+        public Options withMemoryLimits(MemoryLimits memoryLimits) {
+            this.memoryLimits = memoryLimits;
+            return this;
+        }
```

Comment:
with the notable option of WasiOptions (where that however kinda make sense, because it's massive) we were trying to avoid exposing Chicory APIs directly, compare the [go-sdk](https://github.com/extism/go-sdk/blob/b1e59a8b38cb5459ead7dd6f6304213ab8416f64/extism.go#L259-L274)

so, maybe let's add our own MemoryOptions here, or just expose `withMemoryLimits(int,int)` and then instantiate the `MemoryLimits` object here.